### PR TITLE
feat: Feature/legends in analytics

### PR DESCRIPTION
**Pull request by ifoche**
## Description
This PR contains the implementation, using the current code of the app, for the legends in TEI analytics and visualizations analytics as they are implemented in the following SDK PR:
https://github.com/dhis2/dhis2-android-sdk/pull/1733

Please refer to that PR in order to test this PR, in particular the multiple heads-up that are described there regarding how to test it and the limitations that we have found.

## Solution description
The implementation of this PR includes the following main solutions:
- Add legendValue to GraphPoint
- Set LegendValue to GraphPoint
- Render color by legend in tables and single values

## Where did you test this issue?
- [x] Smartphone Emulator
- [ ] Tablet Emulator
- [x] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [x] 9.X - 10.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Review by Balcan — APPROVED** [review]

**Review by andresmr — CHANGES_REQUESTED** [review]
Hi, the code looks great, just some comments to meet requirements related to code style and unit test.
- The code looks good, but you don't meet checkStyle, please run:
`./gradlew dhis_android_analytics:ktlintCheck` to see the issues and solve them.

- `ChartCoordinatesProviderImplTest` is failing, please fix it.

File: dhis_android_analytics/src/main/java/dhis2/org/analytics/charts/di/ChartsInjector.kt [hunk 1]
```diff
@@ -118,9 +119,17 @@ class ChartsModule {
...
+        periodStepProvider: PeriodStepProvider,
+        resourceManager: ResourceManager
     ): ChartCoordinatesProvider {
-        return ChartCoordinatesProviderImpl(d2, periodStepProvider)
+        return ChartCoordinatesProviderImpl(d2, periodStepProvider, resourceManager)
+    }
+
+    @Provides
+    internal fun provideResourceManager(
+        context: Context,
```
[the remark this answers is not in the document]
Comment (ifoche) [reply dhis_android_analytics/src/main/java/dhis2/org/analytics/charts/di/ChartsInjector.kt:130]: @andresmr thanks for your very pertinent comment. We will submit the fix to this shortly.

File: dhis_android_analytics/src/test/java/dhis2/org/analytics/charts/providers/ChartCoordinatesProviderImplTest.kt [hunk 2]
```diff
@@ -15,7 +16,9 @@ import org.mockito.Mockito
 class ChartCoordinatesProviderImplTest {
     private val d2: D2 = Mockito.mock(D2::class.java, Mockito.RETURNS_DEEP_STUBS)
     private val periodStepProvider: PeriodStepProvider = mock()
-    private val coordinatesProvider = ChartCoordinatesProviderImpl(d2, periodStepProvider)
+    private val resourceManager: ResourceManager = mock()
+    private val coordinatesProvider =
+        ChartCoordinatesProviderImpl(d2, periodStepProvider, resourceManager)
 
     @Test
     fun `Should get coordinates for data elements`() {
```
Comment (andresmr) [inline dhis_android_analytics/src/test/java/dhis2/org/analytics/charts/providers/ChartCoordinatesProviderImplTest.kt:24]: These four tests are failing, producing a null pointer exception in `ChartCoordinatesProviderImpl line 47`. Please check it and run the tests.

**Review by andresmr — CHANGES_REQUESTED** [review]
Hi, my previous comments where properly solved, thanks!. The only thing missing is ChartCoordinatesProviderImplTest are failing beacuse of the change of `ChartCoordinatesProviderImpl`, please take a look. thanks!

**Comment by andresmr** [comment]
Hi, please update SDK branch to develop and solve the small conflict. This will be all :). Thanks!

**Pull request merged**